### ISSUE TITLE
.remove does not remove all provided items

### ISSUE THREAD

**Issue by felix-voicehub** [react-native-kit/react-native-track-player#840]
# Configuration

```
System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
    Memory: 2.42 GB / 16.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 12.12.0 - /usr/local/bin/node
    npm: 6.13.6 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-29 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5977832
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    react-native-eject: 0.1.2
```

I'm using v1.2.2.

# Issue
I have the issue (on iOS, not sure about Android) that `.remove([arrayOfTrackIds])` will actually not remove all of the provided track ids from the queue, it seems like with 6 total in the queue and trying to remove 5, it will remove between 2 and 3, and with 12 in the queue and trying to remove 11, it will do 6-8.

I am NOT trying to remove the currently playing track from the queue, I am only talking about tracks after the currently playing one. I am using a draggable list which allows changing the order of the queue, and with below method, my plan is to simply clear everything (except currently playing) from the queue and add it again when the user changes the order.

Any ideas?

# Code

```js
	updateQueue = async (newQueue) => {
		/* Set the new Queue */
		this.setState({ queue: newQueue });

		/* The new queue is all tracks AFTER the currently playing */
		console.log(newQueue.length); // 11

		/* Check the Current Queue (all tracks INCLUDING currently playing) */
		const q1 = await TrackPlayer.getQueue();
		console.log(q1.length); // 12

		/* Now remove the tracks */
		const idsToRemove = newQueue.map((s) => s.id);
		await TrackPlayer.remove(idsToRemove);

		/* Check again */
		const q2 = await TrackPlayer.getQueue();
		/* This should be 12 - 11 = 1 */
		console.log(q2.length); // 6

		/* add */
		await TrackPlayer.add(newQueue);

		/* Check again */
		const q3 = await TrackPlayer.getQueue();
		/* This should be 12 again */
		console.log(q3.length); // 17
	}
```

**Comment by felix-voicehub** [comment]
So I discovered that removing a single item does work consistenly well, so I implemented a workaround of simply removing them in separate promises:

```
		/* Now remove the tracks */
		const removePromises = newQueue.map((s) => TrackPlayer.remove(s.id));
		await Promise.all(removePromises);
```

Still not great as the Docs say providing an array of IDs to remove should be possible, but I think this will get rewritten in v2.0 anyway.

**Comment by JakeMotta** [comment]
I know this issue is closed, but chiming in that I'm having the exact same issue when removing an array of tracks (that aren't the currently playing track). I'll give it a shot removing 1 track at a time though 👍 

Using RNTP v1.2.3.

System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i5-4308U CPU @ 2.80GHz
    Memory: 96.23 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.11.0 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.11.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 23, 25, 26, 27, 28, 29
      Build Tools: 28.0.3, 29.0.3
      System Images: android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.6 AI-192.7142.36.36.6308749
    Xcode: 11.3.1/11C505 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    react-native-cli: 2.0.1

**Comment by curiousdustin** [comment]
Fix released in [1.2.4](https://github.com/react-native-kit/react-native-track-player/releases/tag/v1.2.4)
